HostLibraryTests/DecisionTree_test.cpp

```diff
@@ -369,6 +369,8 @@ TEST(DecisionTree, DecisionTreeBatch)
     using SizeInRange = Predicates::Contraction::SizeInRange;
     using Range       = Predicates::Contraction::Range;
     using And         = Predicates::And<ContractionProblem>;
+    using Key         = std::array<float, 4>;
```

Comment:
I am not sure how Key works, but Key is used at line 374. Isn't it a problem?

Reply:
The dimension of key should be greater than or equal to the number of features (or `features.size()` (however, maybe someone more familiar with solution selection could confirm).

The last test was added along with the additional feature but the `Key` was reused from previous tests where the size was 3. This caused a segfault which is unsuprising because we are trying to access the fourth element of a size three array. What may be a surprise is that this is only an issue on SLES. However, this is UB and the compiler is free to do what it wants. It is possible that this code was optimized out on the other platforms.

As for line 342, we are shadowing the `Key` definition that is at the top of the file. Perhaps we should move these aliases into each of the test cases.